runtime/src/qio/qio_formatted.c

```diff
@@ -763,6 +782,24 @@ qioerr qio_channel_scan_string(const int threadsafe, qio_channel_t* restrict ch,
 
   return err;
 }
+//
+// allocates and returns a string.
+qioerr qio_channel_scan_string(const int threadsafe, qio_channel_t* restrict ch,
+    const char* restrict * restrict out, int64_t* restrict len_out, ssize_t
+    maxlen_bytes)
+{
+  return qio_channel_scan_string_or_bytes(threadsafe, ch, out, len_out,
+                                          maxlen_bytes, 0);
+}
+
+// allocates and returns a string.
+qioerr qio_channel_scan_bytes(const int threadsafe, qio_channel_t* restrict ch,
+    const char* restrict * restrict out, int64_t* restrict len_out, ssize_t
+    maxlen_bytes)
+{
+  return qio_channel_scan_string_or_bytes(threadsafe, ch, out, len_out,
+                                          maxlen_bytes, 1);
+}
```

Comment:
I don't think this is right, because the qio_channel_scan_string_or_bytes code uses read_char and that will have problems with invalid UTF-8 sequences, but these should work for bytes.

Please:
 * copy/paste qio_channel_scan_string to create qio_channel_scan_bytes and change all of the read_char to read_byte
 * add a test of writing a bytes containing an invalid UTF-8 sequence and then reading it back in again, both in binary and text modes